shell_elvish.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"strings"
+)
+
+type elvish struct {
+	re *strings.Replacer
+}
+
+var ELVISH = elvish{}
+
+func (elvish) Hook() (string, error) {
+	return `fn -direnv-hook {
+	direnv export elvish > .direnv.elv
+	-source .direnv.elv
+	/bin/rm .direnv.elv
+}
+
+edit:before-readline = [{ -direnv-hook }]
+`, nil
+}
+
+func (sh elvish) Export(e ShellExport) string {
+	out := ""
+	for k, v := range e {
+		switch v {
+		case nil:
+			out += sh.unset(k)
+		default:
+			out += sh.export(k, *v)
+		}
+	}
+	return out
+}
+
+func (sh elvish) export(k, v string) string {
+	return "E:" + k + " = \"" + sh.escape(v) + "\";\n"
+}
+
+func (sh elvish) unset(k string) string {
+	return "del E:" + k + ";\n"
+}
+
+func (sh elvish) escape(v string) string {
+	return strings.Replace(v, `"`, `\"`, -1)
```

Comment:
I think that you first need to escape `\`. Otherwise `\"` would get escaped to `\\"` instead of `\\\"`

Reply:
The easier way to escape strings in Elvish is to use single quoted strings: Add single quotes at both ends and double the single quotes (replacing `'` with `''`). It is not possible to use `strings.Replace` to build a double-quoted strings because you would need to escape `"` and `\` at the same time.